packages/common/core-utils/src/heap.ts

```diff
@@ -67,7 +67,7 @@ export class Heap<T> {
 	 *
 	 * @returns Heap node containing the smallest element
 	 */
-	public peek(): IHeapNode<T> {
+	public peek(): IHeapNode<T> | undefined {
```

Comment:
Since these are breaking we'll want a changeset so the release notes mention it. Also, let's update the `@returns` section of the docs to call out in which cases would the methods return undefined.

Reply:
I don't think this is a breaking change. The readme of the core-utils calls out that all types in this package are internal (even though this type happens to be marked `@public`). We don't actually have any semver guarantees around this type.

Reply:
I'll be opening a core-utils specific PR to mark everything in the package `@internal`. We can also talk docs there.